Fix linter with latest version of golangci-lint (2.11.4)

With the latest version of golangci-lint, `make checks` fails:

```
golangci-lint run
internal/pgbouncer/management/controller/manager.go:82:14: G118: Long-running loop performs calls without a ctx.Done() cancellation guard (gosec)
			time.Sleep(time.Second)
			          ^
pkg/management/postgres/metrics/histogram/histogram.go:97:39: G115: integer overflow conversion int64 -> uint64 (gosec)
		histogramValue.Buckets[key] = uint64(histogramValue.Values[i])
		                                    ^
pkg/management/postgres/webserver/remote.go:409:3: G118: Goroutine uses context.Background/TODO while request-scoped context is available (gosec)
		go ws.currentBackup.startBackup(context.Background(), &ws.ongoingBackupRequest)
		^
pkg/management/postgres/webserver/remote.go:468:3: G118: Goroutine uses context.Background/TODO while request-scoped context is available (gosec)
		go ws.currentBackup.stopBackup(context.Background(), &ws.ongoingBackupRequest)
		^
internal/controller/cluster_restore_test.go:168:3: Consider preallocating goodPvcs with capacity 3 + len(badPvcs) (prealloc)
		goodPvcs = []corev1.PersistentVolumeClaim{
		^
internal/webhook/v1/cluster_webhook.go:2455:2: Consider preallocating allErrors (prealloc)
	allErrors := field.ErrorList{}
	^
internal/webhook/v1/database_webhook.go:223:6: Consider preallocating allErrs (prealloc)
	var allErrs field.ErrorList
	    ^
7 issues:
* gosec: 4
* prealloc: 3
```

```
golangci-lint --version
golangci-lint has version 2.11.4 built with go1.26.1 from 8f3b0c7e on 2026-03-22T17:35:14Z
```

--- a/internal/controller/cluster_restore_test.go
+++ b/internal/controller/cluster_restore_test.go
@@ -165,7 +165,7 @@ var _ = Describe("getOrphanPVCs", func() {
 			},
 		}
 
-		goodPvcs = []corev1.PersistentVolumeClaim{
+		goodPvcs = []corev1.PersistentVolumeClaim{ //nolint:prealloc
 			{
 				ObjectMeta: metav1.ObjectMeta{
 					Name:      "test-1",

--- a/internal/pgbouncer/management/controller/manager.go
+++ b/internal/pgbouncer/management/controller/manager.go
@@ -79,6 +79,8 @@ func (r *PgBouncerReconciler) Run(ctx context.Context) {
 		if err != nil {
 			// If this is "connection refused" error, it means that apiserver is probably not responsive.
 			// If that's the case wait and resend watch request.
+			//
+			//nolint:gosec
 			time.Sleep(time.Second)
 		}
 	}

--- a/internal/webhook/v1/cluster_webhook.go
+++ b/internal/webhook/v1/cluster_webhook.go
@@ -2452,10 +2452,7 @@ func (v *ClusterCustomValidator) validateManagedRoles(r *apiv1.Cluster) field.Er
 
 // validateManagedExtensions validate the managed extensions parameters set by the user
 func (v *ClusterCustomValidator) validateManagedExtensions(r *apiv1.Cluster) field.ErrorList {
-	allErrors := field.ErrorList{}
-
-	allErrors = append(allErrors, v.validatePgFailoverSlots(r)...)
-	return allErrors
+	return v.validatePgFailoverSlots(r)
 }
 
 func (v *ClusterCustomValidator) validatePgFailoverSlots(r *apiv1.Cluster) field.ErrorList {

--- a/internal/webhook/v1/database_webhook.go
+++ b/internal/webhook/v1/database_webhook.go
@@ -220,7 +220,7 @@ func (v *DatabaseCustomValidator) validateForeignServers(d *apiv1.Database) fiel
 	}
 
 	nameSet := stringset.New()
-	var allErrs field.ErrorList
+	allErrs := make(field.ErrorList, 0, len(d.Spec.Servers))
 	for i, server := range d.Spec.Servers {
 		itemPath := basePath.Index(i)
 

--- a/pkg/management/postgres/metrics/histogram/histogram.go
+++ b/pkg/management/postgres/metrics/histogram/histogram.go
@@ -94,7 +94,16 @@ func NewFromRawData(values []interface{}, columns []string, name string) (*Value
 		if i >= len(histogramValue.Values) {
 			break
 		}
-		histogramValue.Buckets[key] = uint64(histogramValue.Values[i])
+
+		// pq.Array scans into int64 (uint64 is unsupported), so we must cast manually.
+		// Histogram bucket counts are non-negative by definition, so capping at 0
+		// is safe: a negative scan result indicates corruption or overflow, not a
+		// meaningful count.
+		if histogramValue.Values[i] >= 0 {
+			histogramValue.Buckets[key] = uint64(histogramValue.Values[i])
+		} else {
+			histogramValue.Buckets[key] = 0
+		}
 	}
 
 	return histogramValue, nil

--- a/pkg/management/postgres/webserver/remote.go
+++ b/pkg/management/postgres/webserver/remote.go
@@ -406,6 +406,9 @@ func (ws *remoteWebserverEndpoints) backup(w http.ResponseWriter, req *http.Requ
 			sendUnprocessableEntityJSONResponse(w, "CANNOT_INITIALIZE_CONNECTION", err.Error())
 			return
 		}
+
+		// The backup start request continues when the request already terminated.
+		//nolint:gosec
 		go ws.currentBackup.startBackup(context.Background(), &ws.ongoingBackupRequest)
 
 		res := Response[BackupResultData]{
@@ -465,6 +468,8 @@ func (ws *remoteWebserverEndpoints) backup(w http.ResponseWriter, req *http.Requ
 
 		ws.currentBackup.data.Phase = Closing
 
+		// The backup stop request continues when the request already terminated.
+		//nolint:gosec
 		go ws.currentBackup.stopBackup(context.Background(), &ws.ongoingBackupRequest)
 		sendJSONResponseWithData(w, 200, res)
 		return

